### PR TITLE
Create games

### PR DESCRIPTION
**Pull request by mikeymurph77**
I'm not quite sure how to set up the relationship between `Game` and `Team`. 

In the ERD: 
Team has many Games
But does a Game have many Teams? My thoughts: _In a real world situation Yes.... but `Game` is using `home_team_id` and `away_team_id` so I'm thinking that `Game` is not going to recognize a `has_many teams because the many teams are set up by two foreign keys_

To add more reason to my thought's above, when I added the Team has many Games and Games has many Teams tests (with the relationships defined in the models), I got the following output:

```
 # TEST FAILURES
 1) Game should have many teams
     Failure/Error: it { should have_many(:teams) }
         Expected Game to have a has_many association called teams (Team does not have a game_id foreign key.)
 # ./spec/models/game_spec.rb:9:in `block (2 levels) in <top (required)>'

 2) Team should have many games
     Failure/Error: it { should have_many(:games) }
         Expected Team to have a has_many association called games (Game does not have a team_id foreign key.)
 # ./spec/models/team_spec.rb:9:in `block (2 levels) in <top (required)>'
```

### CODE REVIEW

**Comment by edwardloveall** [comment]
Another good question...

The way you've written the ERD is a Game doesn't have Teams at all. It has an away team and a home team. These are both _represented_ by the Team model, but not Teams per se. I might change my answer on this once I see the updated ERD again, but here's how it should work:

``` ruby
class Game < ActiveRecord::Base
  belongs_to :away_team, class_name: 'Team'
  belongs_to :home_team, class_name: 'Team'
end
```

If you haven't seen the options you can pass to associations, there are a lot of them and they're all pretty powerful: http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-belongs_to-label-Options

File: app/models/game.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Game < ActiveRecord::Base
+  validates :week_id, presence: true
+  validates :home_team_id, presence: true
```
Comment (edwardloveall) [inline app/models/game.rb]: Once the associations are in, I believe you can shorten these to `validates :home_team ...`, etc and leave off the id.

File: db/migrate/20160127193752_create_games.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CreateGames < ActiveRecord::Migration
+  def change
+    create_table :games do |t|
+      t.datetime :start_time, null: false
+      t.belongs_to :week, class_name: 'Week', null: false
```
Comment (edwardloveall) [inline db/migrate/20160127193752_create_games.rb:5]: Oh hey look at these options. What do these do in a migration? I'm not sure if they change anything about the database or schema, but I'd love to be proven wrong.

File: spec/models/game_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require 'rails_helper'
+
+describe Game do
+  it { should validate_presence_of(:start_time) }
```
Comment (edwardloveall) [inline spec/models/game_spec.rb]: A lot of times I find it nice to break up association and validation tests with a describe:

``` ruby
describe Game do
  describe 'Associations' do
    # ...
  end

  describe 'Validations' do
    # ...
  end
end
```

File: db/migrate/20160127193752_create_games.rb (hunk 2 above)
Comment (mikeymurph77) [inline db/migrate/20160127193752_create_games.rb:5]: So I googled how to have multiple foreign keys referencing the same table. I came to this [article](http://www.spacevatican.org/2008/5/6/creating-multiple-associations-with-the-same-table/). It's dated back to 2008, so I'm sure there may be a better updated method with Rails 4. 

Pretty much `t.belongs_to :week, class_name: 'Week'` creates the foreign key `week_id` in `Games`.

I'm not sure, but is this doing the same thing that you mentioned in one of your comments.

```
class Game < ActiveRecord::Base
  belongs_to :away_team, class_name: 'Team'
  belongs_to :home_team, class_name: 'Team'
end
```

File: spec/models/game_spec.rb (hunk 3 above)
Comment (mikeymurph77) [inline spec/models/game_spec.rb]: I like this approach. I'll start doing this :+1:

Comment (mikeymurph77) [inline spec/models/game_spec.rb]: Would this be considered a validation test or something like a _method test_ (for lack of a better term)

```
it "returns a user's full name as a string" do
    user = User.new(
      first_name: "Bob",
      last_name: "Willson"
    )

    expect(user.name).to eq 'Bob Willson'
end
```

Comment (edwardloveall) [inline spec/models/game_spec.rb]: I think I'd call it a unit test. The "unit" here is a method.

File: db/migrate/20160127193752_create_games.rb (hunk 2 above)
Comment (edwardloveall) [inline db/migrate/20160127193752_create_games.rb:5]: The difference here is a migration vs a model. That `class_name` code will work fine in a model, but won't do anything in a migration.

**Comment by edwardloveall** [comment]
Other than the migration thing, looks good to go!

**Comment by mikeymurph77** [comment]
@edwardloveall I made the Team association. In order for my tests to work, I needed to specify the class and foreign key on `home_team` and `away_team`

Is this good now to merge?

**Comment by edwardloveall** [comment]
Yeah looks good! Merge away

**Pull request merged**